src/coreclr/pal/src/misc/perfjitdump.cpp

```diff
@@ -314,10 +320,13 @@ struct PerfJitDumpState
             if (!enabled)
                 goto exit;
 
-            result = munmap(mmapAddr, sizeof(FileHeader));
+            if (mmapAddr != NULL)
```

Comment:
I think I'm just missing something, but how does this code ever get executed?  If `enabled` is true, we enter the if condition, but then immediately set it to false.  Then, if `!enabled`, we `goto exit` and skip the call to `munmap`.

Reply:
Uhhh... yes, that is indeed a good catch. Not my fault ;) I've pushed a fix.

Reply:
Thanks.  Totally agree.  Not your fault.  :)

Reply:
(I do not think that the cleanup is required and thus it can be deleted.)

Reply:
I'd be fine deleting the cleanup code here (which appeared to be dead already), and replace it with a comment saying that we are deliberately skipping cleanup for the reasons @jkotas mentioned.

Reply:
I am not able to push the revert of the last commit into your fork. Could you please revert the last commit?

Reply:
@jkotas done! I actually thought about some of this, especially embedding use cases. i actually think either case is fine:
- If we don't do any cleanup, worst case is we leak a few file descriptors for an opt-in perf feature. Could be an issue if you're doing long-running profiling of a process that spins up and shuts down lots of VMs (if that's even possible, not too familiar with the embedding details).
- If we _do_ cleanup and something tries to write from a thread, even if there was no lock, worst case is the write() would return -1 and EBADF. We would hit that FatalError() though, so it might be nice to ignore failures from write and just disable jitdump (enabled = false), but I don't think it's worth it.